Python/codecs.c

```diff
@@ -50,6 +50,26 @@ int PyCodec_Register(PyObject *search_function)
     return -1;
 }
 
+int
+_PyCodec_Unregister(PyObject *search_function)
+{
+    PyInterpreterState *interp = PyInterpreterState_Get();
+    /* Do nothing if codec_search_path is not created yet or was created. */
+    if (interp->codec_search_path == NULL) {
+        return 0;
+    }
+
+    Py_ssize_t n = PyList_Size(interp->codec_search_path);
+    for (Py_ssize_t i = 0; i < n; i++) {
+        PyObject *item = PyList_GetItem(interp->codec_search_path, i);
+        if (item == search_function) {
```

Comment:
PyDict_Clear(interp->codec_search_cache); should only be called here: if the function is found (and removed).

Reply:
ops. you are right.